Server.java

```diff
@@ -3,30 +3,36 @@
 import java.util.Scanner;
 
 public class Server {
-	
+
 	public static void main(String argv[]) throws Exception {
-		
+
 		//Handles clients
 		final ClientHandler clientHandler = new ClientHandler();
+
+		final Log log = new Log(clientHandler.queue);
```

Comment:
So it looks like you're using the clientHandler's queue inside the log? I think we only need either the log or client handler. All the client handler has inside it is the queue.